src/backend/distributed/planner/multi_explain.c

```diff
@@ -1185,6 +1186,17 @@ CitusExplainOneQuery(Query *query, int cursorOptions, IntoClause *into,
 
 	INSTR_TIME_SET_CURRENT(planstart);
 
+	/*
+	 * We should not hide any objects while explaining some query to not break
+	 * postgres vanilla tests.
+	 *
+	 * The filter 'is_citus_depended_object' is added to explain result
+	 * and causes some tests to fail if HideCitusDependentObjects is true.
+	 * Therefore, we disable HideCitusDependentObjects until the current transaction
```

Comment:
maybe also note here why we haven't used `securityQuals` instead of regular quals? I'd assume those wouldn't show up on the EXPLAIN output.